poly/src/evaluations/multivariate/multilinear/sparse.rs

```diff
@@ -227,6 +219,19 @@ impl<F: Field> Index<usize> for SparseMultilinearExtension<F> {
     }
 }
 
+impl<F: Field> Polynomial<F> for SparseMultilinearExtension<F> {
+    type Point = Vec<F>;
+
+    fn degree(&self) -> usize {
```

Comment:
It seems that elsewhere in the crate, we assume that `degree` here represents *total* degree, and not individual degree. This is in agreement with [mathematical convention](https://en.wikipedia.org/wiki/Degree_of_a_polynomial#Extension_to_polynomials_with_two_or_more_variables).

So this should be changed to return `self.num_vars`, and it might be worth adding an `individual_degree` method to `Polynomial` that would return the degree for univariate polynomials, `1` for multilinear polynomials, and the actual individual degree for general multivariate polynomials.

Reply:
I adapted this to represent the total degree. I've also created a separate issue to discuss the exact design and the need for `individual_degree`: https://github.com/arkworks-rs/algebra/issues/692.

Reply:
Is this method meant to represent a total degree _bound_, or the actual total degree of the particular instance this is being called on? For instance, the polynomial `f(x, y, z) = xy + yz` has `num_vars` equal to 3, but actual total degree 2.

Returning the actual total degree would be simple in the sparse case, but might require some work in the dense case if the polynomial is stored in evaluation form.

Incidentally, this caveat would also have applied to the code as it was before @Pratyush 's comment: we were always returning 1, but in the fringe case of constant polynomials, that did not match the total degree.